helios-client/src/main/java/com/spotify/helios/common/JobValidator.java

```diff
@@ -529,4 +523,34 @@ private boolean validateRepositoryName(final String repositoryName,
   private boolean legalPort(final int port) {
     return port >= 0 && port <= 65535;
   }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+
+    private boolean shouldValidateJobHash = true;
+    private boolean shouldValidateAddCapabilities = false;
```

Comment:
I'll get rid of the builder :)

On Thu, Mar 31, 2016 at 4:06 PM, Matt Brown <notifications@github.com>
wrote:

> In helios-client/src/main/java/com/spotify/helios/common/JobValidator.java
> <https://github.com/spotify/helios/pull/896#discussion_r58118888>:
>
> > @@ -529,4 +523,34 @@ private boolean validateRepositoryName(final String repositoryName,
> >    private boolean legalPort(final int port) {
> >      return port >= 0 && port <= 65535;
> >    }
> > +
> > +  public static Builder newBuilder() {
> > +    return new Builder();
> > +  }
> > +
> > +  public static class Builder {
> > +
> > +    private boolean shouldValidateJobHash = true;
> > +    private boolean shouldValidateAddCapabilities = false;
>
> if you really want to make this optional (and I don't think it needs to
> be), it should at least default to true, so that it isn't accidentally left
> in the off position
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/spotify/helios/pull/896/files/6193a4dc713e6138693cb27dd2151fb0a105acaf#r58118888>
>



-- 
David Xia
Software Engineer
dxia@spotify.com